docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Deploy using Helm.
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Not implemented.
```

Comment:
`Not implemented` indicates a WIP state for me. As if we haven't built it yet. But in general the `reset` and `clean` of components like KubernetesApp are skipped.

Reply:
Maybe we should not mention them or explicitly write that we are skipping these steps.

Reply:
Changed them to "Do nothing."